### ISSUE TITLE
Bump dependency System.Configuration.ConfigurationManager to 6.0.0 to avoid vulnerability issues with System.Drawing.Common 4.7.0

### ISSUE THREAD

**Issue by sander1095** [quartznet/quartznet#1792]
Currently this project has a dependency on `System.Configuration.ConfigurationManager` version `4.7.0`

In a project I work on we use Trivy for container scanning purposes. This reported issue [CVE-2021-24112](https://msrc.microsoft.com/update-guide/vulnerability/CVE-2021-24112) because of a dependency on `System.Drawing.Common`  version `4.7.0`.

```
┌───────────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬─────────────────────────────────────────────┐
│        Library        │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                    Title                    │
├───────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼─────────────────────────────────────────────┤
│ System.Drawing.Common │ CVE-2021-24112 │ CRITICAL │ 4.7.0             │ 5.0.3, 4.7.2  │ dotnet: Remote Code Execution Vulnerability │
│                       │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2021-24112  │
└───────────────────────┴────────────────┴──────────┴───────────────────┴───────────────┴─────────────────────────────────────────────┘
```

After a while, I figured out that we had this dll because of Quartz. 
Quartz has this dependency because of `System.Configuration.ConfigurationManager`

This issue is fixed in `5.0.3`, `4.7.2`  (or `6.x.x`).

I recommend updating `SYstem.ConfigurationManager` to the latest `6.x.x` version so this vulnerability will be fixed.

**Comment by lahma** [comment]
I don't think `System.Configuration.ConfigurationManager` has dependency on `System.Drawing.Common`, what Quartz packages and their versions are you actually using?

**Comment by sander1095** [comment]
You can see here that `System.Configuration.ConfigurationManager` 4.7.0 has a dependency on `System.Security.Permissions` 4.7.0:

https://www.nuget.org/packages/System.Configuration.ConfigurationManager/4.7.0#dependencies-body-tab

And here you can see that `System.Security.Permissions` has a dependency on `https://www.nuget.org/packages/System.Security.Permissions/4.7.0#dependencies-body-tab` (in some cases)

https://www.nuget.org/packages/System.Security.Permissions/4.7.0#dependencies-body-tab

And here you can see that `System.Windows.Extensions` has a dependency on `System.Drawing.Common`:

https://www.nuget.org/packages/System.Windows.Extensions/4.7.0#dependencies-body-tab

**Comment by lahma** [comment]
Wow, that's a weird dependency tree. Thanks for sharing!

**Comment by lahma** [comment]
Went with the minimal increment that fixes the vulnerable dependency,  5.0.0. Available on 3.x branch in MyGet feed shortly, thanks for report and guidance!

**Comment by sander1095** [comment]
Hi @lahma , I don't think this is good enough, sadly!

If you follow the breadcrumbs, you can see that `System.Windows.Extensions` 5.0.0 still has a dependency on `System.Drawing.Common` 5.0.0, which still contains the vulnerability!

https://www.nuget.org/packages/System.Windows.Extensions/5.0.0#dependencies-body-tab

The vulnerability is fixed in 5.0.3!
I am not sure how nuget works with retrieving a dependency's dependency's dependency, but my guess is that it will retrieve 5.0.0. In any case, if someone would use `5.0.0`, `5.0.1` or `5.0.2`, the dependency chain would still be valid, but the vulnerability would remain.

So to be sure, I see 2 options:

1. If the user is targetting `netcoreapp3.0`, Quartz should explicitly install `System.Drawing.Common` 5.0.3 to work around this issue, OR
2. We should upgrade to `System.Configuration.ConfigurationManager` 6.0.0 so the dependency tree has a dependency on `System.Drawing.Common` 6.0.0 where this is explicitly fixed

I am a bigger fan of option 2, because I think it's weird for quartz to introduce a direct dependency on `System.Drawing.Common`.. 

I am curious to hear what you think. I of course don't want to decrease the surface area of this library either :)

**Comment by lahma** [comment]
OK now we're at version  6.0.1 at 3.x branch I'll remove the whole dependency on main (4.x).
